File: KoreanTranslationCamera/KoreanTranslationCamera/Views/LanguageSelectorView.swift
import SwiftUI

// MARK: - Language Selector
//
// Horizontal row of language pills shown at the top of the screen.
// The active language is highlighted; tapping another pill switches immediately.

struct LanguageSelectorView: View {

    let selectedLanguage: SupportedLanguage
    let onSelect: (SupportedLanguage) -> Void

    var body: some View {
        HStack(spacing: 10) {
            ForEach(SupportedLanguage.allCases) { language in
                LanguagePill(
                    language: language,
                    isSelected: language == selectedLanguage
                ) {
                    onSelect(language)
                }
            }
        }
        .padding(.horizontal, 16)
        .padding(.vertical, 10)
        .background(.ultraThinMaterial, in: Capsule())
    }
}

// MARK: - Language Pill

private struct LanguagePill: View {

    let language: SupportedLanguage
    let isSelected: Bool
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            Text(language.displayName)
                .font(.system(size: 14, weight: isSelected ? .semibold : .regular))
                .foregroundStyle(isSelected ? Color.black : Color.white)
                .padding(.horizontal, 16)
                .padding(.vertical, 8)
                .background(
                    isSelected ? Color.white : Color.white.opacity(0.15),
                    in: Capsule()
                )
        }
        .buttonStyle(.plain)
        .animation(.easeInOut(duration: 0.15), value: isSelected)
    }
}

#Preview {
    ZStack {
        Color.black.ignoresSafeArea()
        LanguageSelectorView(selectedLanguage: .english) { _ in }
    }
}

File: KoreanTranslationCameraAndroid/app/src/main/java/com/personal/koreantranslation/ui/LanguageSelectorView.kt
package com.personal.koreantranslation.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.personal.koreantranslation.models.SupportedLanguage

// ---------------------------------------------------------------------------
// Language Selector
//
// Horizontal row of language pills at the top of the screen.
// ---------------------------------------------------------------------------

@Composable
fun LanguageSelectorView(
    selectedLanguage: SupportedLanguage,
    onSelect: (SupportedLanguage) -> Unit,
    modifier: Modifier = Modifier
) {
    Row(
        modifier = modifier
            .background(
                color = Color.White.copy(alpha = 0.15f),
                shape = RoundedCornerShape(50)
            )
            .padding(horizontal = 8.dp, vertical = 6.dp),
        horizontalArrangement = Arrangement.spacedBy(6.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        SupportedLanguage.entries.forEach { language ->
            LanguagePill(
                language     = language,
                isSelected   = language == selectedLanguage,
                onClick      = { onSelect(language) }
            )
        }
    }
}

@Composable
private fun LanguagePill(
    language: SupportedLanguage,
    isSelected: Boolean,
    onClick: () -> Unit
) {
    Button(
        onClick = onClick,
        shape   = CircleShape,
        colors  = ButtonDefaults.buttonColors(
            containerColor = if (isSelected) Color.White else Color.Transparent,
            contentColor   = if (isSelected) Color.Black  else Color.White
        ),
        elevation = null
    ) {
        Text(
            text       = language.displayName,
            fontSize   = 14.sp,
            fontWeight = if (isSelected) FontWeight.SemiBold else FontWeight.Normal
        )
    }
}
